src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1993,7 +2189,11 @@ internal void GenerateImplicitConversionErrorForCollectionExpression(
                         reportedErrors = true;
                     }
 
-                    if (elements.Length > 0 &&
+                    if (GetCollectionExpressionApplicableIndexer(node.Syntax, targetType, elementTypeWithAnnotations.Type, BindingDiagnosticBag.Discarded) is { })
```

Comment:
>BindingDiagnosticBag.Discarded [](http://example.com/codeflow?start=123&length=30)

Feels like we should collect the diagnostics/use-site-info from binding the indexer (rather than just dropping them) and add them to `diagnostics` if we move forward with indexer.

Reply:
Added PROTOTYPE comment.